### DISCUSSION TITLE
How would I make hexagonal widgets?

### DISCUSSION THREAD

**Discussion by veniamin-ilmer** [hecrj/iced#2037] [Q&A]
I'd want to build games like hexagonal chess, hexagonal minesweeper, etc, so I was interested in hexagonal widget, hexagonal tiling, etc.

From what I see, all widgets seem to boil down to using `renderer::Quad`.

Checking out [renderer](https://docs.rs/iced/latest/iced/advanced/renderer/index.html), I see it only had Quad. No other shapes available. 

Even the "custom widget" circle example uses `Quad`, except with rounded edges. I don't even know how you'd determine mouse interaction, being able to tell where is the inside of the rounded edges.

I got excited about the [integration example](https://github.com/iced-rs/iced/tree/master/examples/integration), because it features a triangle, but it seems like the triangle is specific to wgpu, using `wgpu::PrimitiveTopology::TriangleList`.

It seems if I want to build anything hexagonal, I'd need to do a bunch of math myself with lots of lines, correct?

**Comment by genusistimelord** [comment]
you can use the Canvas to render anything you want.

https://docs.rs/iced/latest/iced/widget/canvas/index.html
https://docs.rs/iced/latest/iced/widget/struct.Canvas.html

**Reply by veniamin-ilmer** [reply]
So I'd need to do a bunch of math myself with lots of lines, correct?

**Reply by genusistimelord** [reply]
Correct. also due to the layout system you might have to do most things this way since we cant take into effect its size or multiple sides. which would cause odd spacing etc. so you can just use the canvas as a way to render them out fully.
